hazelcast-jclouds/src/main/java/com/hazelcast/jclouds/ComputeServiceBuilder.java

```diff
@@ -114,27 +133,32 @@ public int getServicePort() {
         return getOrDefault(JCloudsProperties.HZ_PORT, NetworkConfig.DEFAULT_PORT);
     }
 
-    private boolean checkRegionZoneConfig(NodeMetadata metadata) {
+    public boolean isNodeInsideZones(NodeMetadata metadata) {
         Location location = metadata.getLocation();
         while (location != null) {
             String id = location.getId();
-            switch (location.getScope()) {
-                case ZONE:
-                    if (id != null && regionsSet.contains(id)) {
-                        return true;
-                    }
-                    break;
-                case REGION:
-                    if (id != null && !zonesSet.contains(id)) {
-                        return true;
-                    }
-                    break;
-                default:
-                    break;
+            if (location.getScope().equals(LocationScope.ZONE)) {
+                if (id != null && !zonesSet.isEmpty() && !zonesSet.contains(id)) {
+                    return false;
+                }
             }
             location = location.getParent();
         }
-        return false;
+        return true;
+
+    }
+    public boolean isNodeInsideRegions(NodeMetadata metadata) {
+        Location location = metadata.getLocation();
+        while (location != null) {
+            String id = location.getId();
+            if (location.getScope().equals(LocationScope.REGION)) {
+                if (id != null && !regionsSet.isEmpty() && !regionsSet.contains(id)) {
+                    return false;
+                }
+            }
+            location = location.getParent();
+        }
+        return true;
```

Comment:
Yes, it might be feel strange, but a `Node` in jclouds  always has `REGION` scope, because you cannot define a `ComputeResource` without binding it to a region or zone. So your case will never happen.